kroxylicious-kubernetes-api/src/main/java/io/kroxylicious/kubernetes/api/common/TrustAnchorRef.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.api.common;
+
+import io.fabric8.kubernetes.api.model.KubernetesResource;
+
+/**
+ * A reference, used in a kubernetes resource, to a resource containing trust anchor(s).
+ */
+@com.fasterxml.jackson.annotation.JsonInclude(com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL)
+@com.fasterxml.jackson.annotation.JsonPropertyOrder({ "group", "kind", "name", "key" })
+@com.fasterxml.jackson.databind.annotation.JsonDeserialize(using = com.fasterxml.jackson.databind.JsonDeserializer.None.class)
+@javax.annotation.processing.Generated("io.fabric8.java.generator.CRGeneratorRunner")
+@lombok.ToString()
+@io.sundr.builder.annotations.Buildable(editableEnabled = false, validationEnabled = false, generateBuilderPackage = false, builderPackage = "io.fabric8.kubernetes.api.builder", refs = {
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.ObjectMeta.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.ObjectReference.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.LabelSelector.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.Container.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.EnvVar.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.ContainerPort.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.Volume.class),
+        @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.VolumeMount.class)
+})
+public class TrustAnchorRef
+        extends AbstractLocalRef
```

Comment:
This seems dodgy to me. `LocalRef` defines exactly what equality means for its subclasses, and that requires that two TrustAnchorRefs with different `key` but the same other properties are considered equal. I can see that we probably don't depend on the equality semantics for what's happening in this PR, but it still seems like it's just going to wait in the code base and surprise someone in the future. 

Perhaps we can get away with TrustAnchorRef having a `LocalRef`-typed property?

Reply:
@tombentley @robobario please take another look.

Inspired by https://github.com/kubernetes-sigs/gateway-api/blob/1e14dd602df6c0bad62c0e7f83dd2c7d1d2d6886/apis/v1alpha2/policy_types.go#L85 I have used ` @JsonUnwrapped` so that the `AnyLocalRef` property of TrustAnchorRef is inlined into the `trustAnchorRef` serialised form.  This means the API is unchanged. WDYT?